src/Command/ValidateCommand.php

```diff
@@ -44,30 +52,31 @@ protected function doExecute(DependencyGraph $graph): int
         ]));
 
         $violations = $validator->getViolations($graph);
+        $filteredViolations = $this->filterViolations($graph, $violations);
 
-        if ($violations->isEmpty()) {
+        if ($filteredViolations->isEmpty()) {
             $this->io->success(sprintf('Project located at "%s" is valid.', $graph->getRootPath()));
 
             return 0;
         }
 
-        foreach ($violations as $violation) {
+        foreach ($filteredViolations as $violation) {
             $this->io->writeln(sprintf('<violation>[VIOLATION]</violation> %s', $violation->getFormattedMessage()));
         }
 
-        if (!empty($violations)) {
+        if (!empty($filteredViolations)) {
```

Comment:
Unnecessary condition as you are already checking this on line 57 and returning if so.